Do not add SharedMount if it is invalid

rullzer opened the pull request:
If a shared file is moved to the trash the share entry is still alive.
However the sharee can't initialise the share since it is not in the
owners dir <user>/files.

So check if it is there and if not log a debug and carry on.

Fixes #938 

@icewind1991 This is safe right (we do it later in the code otherwise anyways that is where the error comes from). But this will not trigger depenency setup right?

CC: @LukasReschke @nickvergessen @MorrisJobke @blizzz

rullzer commented:
Nope I was to quick...

rullzer commented:
@icewind1991 maybe you have a 'somewhat' clean solution. Basically it fails in https://github.com/nextcloud/server/blob/master/apps/files_sharing/lib/sharedstorage.php#L87 because the file is in the trash (so not in `<user>/files`).

nickvergessen commented on apps/files_sharing/lib/SharedMount.php:
```diff
@@ -72,7 +72,12 @@ public function __construct($storage, array $mountpoints, $arguments = null, $lo
+		$ownerView = new View('/' . $this->superShare->getShareOwner() . '/files');
+		$arguments['ownerView'] = $ownerView;
+
+		// Check if the fileid is in the owners home (and not in the trash) see #938
```
I'd prefer full link to avoid problems with numbers of another project

rullzer committed:
Do not error on NotFoundException

Since this exception can be thrown when a shared file is in the
trashbin there is no need to spam the log like crazy.

Fixes #938

rullzer commented:
Rebased on @icewind1991 branch. Lets see what CI tells us.

derkostka commented:
Maybe also fixes #710

derkostka commented:
👍 works as expected. Changes look good.

LukasReschke commented:
Seems to do 👍

MorrisJobke commented:
@rullzer Could you create the backport PR?

rullzer commented:
Ah yes forgot. will do

rullzer commented:
Ah no I did https://github.com/nextcloud/server/pull/1029

MorrisJobke commented:
> Ah no I did #1029

Ah got it ... it was merged into a different branch and not into master.
